seatunnel-connectors/seatunnel-connector-flink-socket/src/main/java/org/apache/seatunnel/flink/connector/socket/source/SocketSourceFunction.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.flink.connector.socket.source;
+
+import org.apache.flink.api.common.serialization.DeserializationSchema;
+import org.apache.flink.api.common.serialization.RuntimeContextInitializationContextAdapters;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.java.typeutils.ResultTypeQueryable;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.streaming.api.functions.source.RichSourceFunction;
+import org.apache.flink.table.data.RowData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+
+/**
+ * The {@link SocketSourceFunction} opens a socket and consumes bytes.
+ *
+ * <p>It splits records by the given byte delimiter (`\n` by default) and delegates the decoding to
+ * a pluggable {@link DeserializationSchema}.
+ *
+ * <p>Note: This is only an example and should not be used in production. The source function is not
+ * fault-tolerant and can only work with a parallelism of 1.
+ */
+public final class SocketSourceFunction extends RichSourceFunction<RowData>
+        implements ResultTypeQueryable<RowData> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SocketSourceFunction.class);
+
+    private final String hostname;
+    private final int port;
+    private final byte byteDelimiter;
+    private final DeserializationSchema<RowData> deserializer;
+    private static final int WAIT_TIME = 1000;
+
+    private volatile boolean isRunning = true;
+    private Socket currentSocket;
```

Comment:
It is not recommended to use the Socket filed that cannot be serialized in the Serializable class

Reply:
It seems a minor problem, do you have good practice if we don't use `Socket`  here?

Reply:
We need define the variable here, and will use it in `cancel` method.